tests/conftest.py

```diff
@@ -1,84 +1,166 @@
+    computer.configure(
+        url=firecrest_config.url,
+        token_uri=firecrest_config.token_uri,
+        client_id=firecrest_config.client_id,
+        client_secret=firecrest_config.client_secret,
+        compute_resource=firecrest_config.compute_resource,
+        small_file_size_mb=firecrest_config.small_file_size_mb,
+        temp_directory=firecrest_config.temp_directory,
+        api_version=firecrest_config.api_version,
     )
+    computer.store()
+    return computer
+
+
+class MockFirecrest:
+    def __init__(self, firecrest_url, *args, **kwargs):
+        self._firecrest_url = firecrest_url
+        self.args = args
+        self.kwargs = kwargs
+
+        self.whoami = whoami
+        self.list_files = list_files
+        self.stat = stat_
+        self.mkdir = mkdir
+        self.simple_delete = simple_delete
+        self.parameters = parameters
+        self.symlink = symlink
+        self.checksum = checksum
+        self.simple_download = simple_download
+        self.simple_upload = simple_upload
+        self.compress = compress
+        self.extract = extract
+        self.copy = copy
+        self.submit = submit
+        self.poll_active = poll_active
+
+
+class MockClientCredentialsAuth:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+
+@dataclass
+class ComputerFirecrestConfig:
+    """Configuration of a computer using FirecREST as transport plugin."""
+
+    url: str
+    token_uri: str
+    client_id: str
+    client_secret: str
+    compute_resource: str
+    temp_directory: str
+    workdir: str
+    api_version: str
+    small_file_size_mb: float = 1.0
+
+
+class RequestTelemetry:
+    """A to gather telemetry on requests."""
+
+    def __init__(self) -> None:
+        self.counts = {}
+
+    def wrap(
+        self,
+        method: Callable[..., requests.Response],
+        url: str | bytes,
+        **kwargs: Any,
+    ) -> requests.Response:
+        """Wrap a requests method to gather telemetry."""
+        endpoint = urlparse(url if isinstance(url, str) else url.decode("utf-8")).path
+        self.counts.setdefault(endpoint, 0)
+        self.counts[endpoint] += 1
+        return method(url, **kwargs)
 
 
 @pytest.fixture(scope="function")
-def firecrest_server(
+def firecrest_config(
     pytestconfig: pytest.Config,
     request: pytest.FixtureRequest,
     monkeypatch,
     tmp_path: Path,
 ):
-    """A fixture which provides a mock Firecrest server to test against."""
+    """
+    If a config file is given it sets up a client environment with the information
+    of the config file and uses pyfirecrest to communicate with the server.
+    ┌─────────────────┐───►┌─────────────┐───►┌──────────────────┐
+    │ aiida_firecrest │    │ pyfirecrest │    │ FirecREST server │
+    └─────────────────┘◄───└─────────────┘◄───└──────────────────┘
+
+    if `config_path` is not given, it monkeypatches pyfirecrest so we never
```

Comment:
```suggestion
    if no config file is not provided, it monkeypatches pyfirecrest so we never
```

Reply:
The suggestion is an unclear sentence :)